### ISSUE TITLE
Differences among model variants

### ISSUE THREAD

**Issue by poohhsu** [yformer/EfficientTAM#12]
Thank you for sharing your work! I am currently using your models and noticed several variations:

- efficienttam_s.pt
- efficienttam_s_1.pt
- efficienttam_s_2.pt
- efficienttam_s_512x512.pt

I also observed that some of these models use EfficientRoPEAttention1 and EfficientRoPEAttention2. Could you please clarify the differences between them and the differences among these models? Thank you!

**Comment by yformer** [comment]
Hi @poohhsu, thanks for your interest in our work. 

efficienttam_s.pt uses 1024x1024 input resolution.
efficienttam_s_512x512.pt uses 512x512 input resolution.

EfficientRoPEAttention2 and EfficientRoPEAttention1 are efficient memory cross attention variants, which reduces the attention complexity by pooling spatial tokens. 
efficienttam_s_1.pt uses EfficientRoPEAttention1 for efficient memory attention.
efficienttam_s_2.pt uses EfficientRoPEAttention2 for efficient memory attention.

**Comment by Zarxrax** [comment]
Can you elaborate more on what the EfficientRoPEAttention means in practical terms? Do they provide the same quality of output as the standard version, or are they just faster? By what metrics would one choose one of these models over the other?

I did some benchmarks for propagation across a 200 frame, half-hd clip, with the following results (RTX 2060 12GB):
```
sam2.1baseplus: 2m21s
efficienttam_s: 2m03s
efficienttam_s_1: 1m06s
efficienttam_s_2: 1m05s
efficienttam_s_512x512: 0m17s
```
I was initially quite disappointed by the performance of the standard efficienttam_s model, since it was not much faster than regular sam2. But the _1 & _2 versions are in line with the charts you have provided. The speed of the 512x512 version blew me away, though I recognize there is a slight quality penalty on that one.

**Comment by yformer** [comment]
Hi @Zarxrax, thanks for your interest in our work!

EfficientRoPEAttention means efficient memory cross attention, which will improve the efficiency with a marginal quality penalty. If you have need for efficiency while maintaining high quality performance, we recommend EfficientRoPEAttention over the standard version. 

We benchmarked models on A100 using [the benchmark script](https://github.com/yformer/EfficientTAM/blob/main/efficient_track_anything/benchmark.py) w.r.t. FPS,
```
sam2.1baseplus: 62.7 FPS
sam2baseplus: 63.3 FPS
sam2small: 84.4 FPS
small2tiny: 90 FPS
efficienttam_s: 100.4 FPS
efficienttam_s_1: 106.4 FPS
efficienttam_s_2: 124.5 FPS
efficienttam_s_512x512: 270.9 FPS
```
I think the efficiency will depend on the hardware you are running. We did not have RTX 2060 and other RTX series for benchmarking these models. Thanks for adding one data point here. 

If you want to push for efficiency over quality, (e.g., running on Mac/iPhone 15 with > 10 FPS), with ~3% accuracy drop w.r.t. EfficientTAM_S, we recommend efficienttam_s_512x512.

Let me know if you have other questions.

**Comment by okideal** [comment]
@yformer Hello author. Does the export of models with other output sizes require additional fine-tuning? For example, if I want to change the input to 960x540 to adapt to the aspect ratio of 1080p.

**Comment by yformer** [comment]
> [@yformer](https://github.com/yformer) Hello author. Does the export of models with other output sizes require additional fine-tuning? For example, if I want to change the input to 960x540 to adapt to the aspect ratio of 1080p.

@okideal, it does not require additional finetuning.

**Comment by whistle-blower-kingdom** [comment]
@yformer 
The FPS in your [responce](https://github.com/yformer/EfficientTAM/issues/12#issuecomment-2571813689) and paper are measured under vos_compiled=True？yes or no

**Comment by laisimiao** [comment]
@yformer what is their each performance? like efficienttam_s vs. efficienttam_s_2?
